fw/str.h

```diff
@@ -239,6 +239,8 @@ basic_stricmp_fast(const BasicStr *s1, const BasicStr *s2)
 #define TFW_STR_NOCCPY_HDR	0x20
 /* Weak identifier was set for Etag value. */
 #define TFW_STR_ETAG_WEAK	0x40
+/* Etag value isn't enclosed in double quotes */
+#define TFW_STR_ETAG_NO_DOUBLE_QUOTES 0x80
```

Comment:
I see that you introduced the flag because there is already `TFW_STR_ETAG_WEAK`. Apparently it was stored in the `TfwStr` descriptor of the header for (efficient) strong matching. However, according to RFC 7234 4.3.2 caches deal with `If-None-Match` and `If-Modified-Since` only, so use weak matching only (see `tfw_handle_validation_req()`).

So I believe storing both of the flags in `TfwStr` is wrong. After all, `TfwStr` is a very generic string descriptor, not only for the etag values and we need to have good reasons to introduce very specific flags to the very generic layer.

Both of the flags are used for the parser only and should be handler only there, probably in `TfwHttpParser->flags`

Reply:
I fix it for TFW_STR_ETAG_NO_DOUBLE_QUOTES, but i don't agree with you about TFW_STR_ETAG_WEAK. For 'If-Match' header we should use a strong comparison. I don't know why we don't check this header, but later when we implement it we need to store this flag for strong matching.

Reply:
Fixed, also remove TFW_STR_ETAG_WEAK at all